certificatesHandler.go

```diff
@@ -3,96 +3,114 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"math/rand"
 	"net/http"
-	"strconv"
-	"time"
 
 	"github.com/gorilla/mux"
 )
 
-// GetUsersCertificates route handler get user's certificates method
-func (a *App) GetUsersCertificates(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	params := mux.Vars(r)
-	//Authenticate user and pw
-	userEmail, _, _ := r.BasicAuth()
-	var usersCertificates []Certificate
-	for _, item := range certificates {
-		userID, err := LookUpUserIDByEmail(userEmail)
-		if err != nil {
-			fmt.Fprintf(w, err.Error())
-			return
-		}
-		//Check the correct user is attempting to view the page
-		if item.OwnerID == userID && params["userId"] == userID {
-			usersCertificates = append(usersCertificates, item)
-		}
-		if userID != params["userId"] {
-			fmt.Fprintf(w, "You cannot access user "+params["userId"]+"'s certificates")
-			return
-		}
-	}
-	if err := json.NewEncoder(w).Encode(usersCertificates); err != nil {
-		fmt.Fprintf(w, err.Error())
-	}
-}
-
 // CreateCertificate route handler POST certificate method
 func (a *App) CreateCertificate(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
+	userName, _, _ := r.BasicAuth()
+	userID, err := LookUpUserIDByName(userName)
+	if err != nil {
+		fmt.Fprintf(w, err.Error())
+		return
+	}
 	var certificate Certificate
-	userEmail, _, _ := r.BasicAuth()
+	var rc Certificate
 	defer r.Body.Close()
-	if err := json.NewDecoder(r.Body).Decode(&certificate); err != nil {
+	if err := json.NewDecoder(r.Body).Decode(&rc); err != nil {
 		fmt.Fprintf(w, err.Error())
 		return
 	}
-	if certificate.Title == "" {
+	if rc.Title == "" {
 		w.WriteHeader(http.StatusBadRequest)
 		json.NewEncoder(w).Encode(map[string]interface{}{
 			"title": "Title must be populated",
 		})
 		return
 	}
-
-	certificate.OwnerID, _ = LookUpUserIDByEmail(userEmail)
-	// Todo (Farouk): Mock ID - not safe
```

Comment:
It's fine for an example piece of work. If you was using an actual database then indexing would be covered. So it's good that time was not wasted on this.